File: java/src/test/java/com/jamesansley/ValidParenthesesTest.java
package com.jamesansley;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class ValidParenthesesTest {
    @Test
    public void exampleCases() {
        ValidParentheses solution = new ValidParentheses();

        assertTrue(solution.isValid("()"));
        assertTrue(solution.isValid("()[]{}"));
        assertTrue(solution.isValid("([])"));
        assertFalse(solution.isValid("(]"));
    }
}

File: javascript/tests/valid_parentheses_test.js
import {assert, assertFalse} from "@std/assert";
import {isValid} from "@/valid_parentheses.js";

Deno.test(function exampleCases() {
    assert(isValid("()"));
    assert(isValid("()[]{}"));
    assert(isValid("([])"));
    assertFalse(isValid("(]"));
});

Deno.test(function extraCases() {
    assertFalse(isValid("(()"));
    assertFalse(isValid("())"));
    assertFalse(isValid("]["));
});
